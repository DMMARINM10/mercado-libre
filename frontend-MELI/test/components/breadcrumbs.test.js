import React from 'react';
import { render, screen } from '@testing-library/react';
import '@testing-library/jest-dom';
import Breadcrumb from '@/components/Breadcrumbs';

describe('Breadcrumb', () => {
  const props = {
    breadcrumb: ['Category 1', 'Category 2', 'Category 3']
  };

  it('should render the breadcrumb with correct categories and icons', () => {
    render(<Breadcrumb {...props} />);

    const breadcrumbItems = screen.getAllByRole('listitem');
    expect(breadcrumbItems).toHaveLength(props.breadcrumb.length);
    
    breadcrumbItems.forEach((item, index) => {
        const textContent = item.textContent.trim();
        expect(textContent).toBe(props.breadcrumb[index]);
    });

    const icons = screen.queryAllByTestId('ArrowForwardIosIcon');
    expect(icons).toHaveLength(props.breadcrumb.length - 1);
  });

  it('should match the snapshot', () => {
    const { asFragment } = render(<Breadcrumb {...props}/>);
    expect(asFragment()).toMatchSnapshot();
  });
});
