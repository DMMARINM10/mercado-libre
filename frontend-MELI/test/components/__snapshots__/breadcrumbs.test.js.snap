// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Breadcrumb should match the snapshot 1`] = `
<DocumentFragment>
  <nav>
    <ul
      class="breadcrumb"
    >
      <li>
        Category 1
        <svg
          aria-hidden="true"
          class="MuiSvgIcon-root MuiSvgIcon-fontSize14px css-821wyw-MuiSvgIcon-root"
          data-testid="ArrowForwardIosIcon"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M6.23 20.23 8 22l10-10L8 2 6.23 3.77 14.46 12z"
          />
        </svg>
      </li>
      <li>
        Category 2
        <svg
          aria-hidden="true"
          class="MuiSvgIcon-root MuiSvgIcon-fontSize14px css-821wyw-MuiSvgIcon-root"
          data-testid="ArrowForwardIosIcon"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M6.23 20.23 8 22l10-10L8 2 6.23 3.77 14.46 12z"
          />
        </svg>
      </li>
      <li>
        Category 3
      </li>
    </ul>
  </nav>
</DocumentFragment>
`;
